gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/javaversions/BaselineJavaVersion.java

```diff
@@ -88,11 +99,18 @@ public void execute(CheckJavaVersionsTask task) {
                     });
 
             TaskProvider<CheckClasspathCompatible> checkRuntimeClasspathCompatible = project.getTasks()
-                    .register("checkRuntimeClasspathCompatible", CheckClasspathCompatible.class, task -> {
-                        task.getClasspathName().set("runtime");
-                        task.getJavaVersion().set(extension.runtime());
-                        task.getClasspath().setFrom(project.getConfigurations().getByName("runtimeClasspath"));
-                    });
+                    .register(
+                            "checkRuntimeClasspathCompatible",
+                            CheckClasspathCompatible.class,
+                            new Action<CheckClasspathCompatible>() {
```

Comment:
Register is fine for lambas, it's only task actions/CommandLineArgumentProviders that need the lambda